pkg/controller/podgc/gc_controller_test.go

```diff
@@ -448,3 +451,149 @@ func TestGCUnscheduledTerminating(t *testing.T) {
 		})
 	}
 }
+
+func TestGCTerminating(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.NonGracefulFailover, true)()
+	type node struct {
+		name           string
+		readyCondition v1.ConditionStatus
+		taints         []v1.Taint
+	}
+
+	type nameToPodConfig struct {
+		name              string
+		phase             v1.PodPhase
+		deletionTimeStamp *metav1.Time
+		nodeName          string
+	}
+
+	testCases := []struct {
+		name            string
+		pods            []nameToPodConfig
+		nodes           []node
+		deletedPodNames sets.String
+	}{
+		{
+			name: "pods have deletion timestamp set and the corresponding nodes are not ready",
+			nodes: []node{
+				{name: "worker-0", readyCondition: v1.ConditionFalse},
+				{name: "worker-1", readyCondition: v1.ConditionFalse},
+			},
+			pods: []nameToPodConfig{
+				{name: "a", deletionTimeStamp: &metav1.Time{}, nodeName: "worker-0"},
+				{name: "b", deletionTimeStamp: &metav1.Time{}, nodeName: "worker-1"},
+			},
+			deletedPodNames: sets.NewString(),
+		},
+
+		{
+			name: "some pods has deletion timestamp and/or phase set and some of the corresponding nodes have " +
```

Comment:
```suggestion
			name: "some pods have deletion timestamp and/or phase set and some of the corresponding nodes have an" +
```

Reply:
I'd also maybe just have this on one line - or break it into multiple cases. This case is a little tricky to skim.

Reply:
Agreed. I thought to break this one and then finally went via this route. 
My reasoning: 
I wanted to create a env where I have couple of pods distributed over couple of nodes. Plus, the nodes can be either `Ready` on `NotReady` and pods could be in diff phase. 

( Just to note that -- pod phase comes into picture -- due to the fact that we want to check if the pod is terminating or not. So we are saying the pod is terminating only when deletion timestamp is set and do not care what the phase is. There were some thoughts earlier if we want to check for pod phase or not. Currently we do not care the pod phase. )

As I was testing the top level function, `gcTerminating` -- I thought it would make sense to make it act on such env as a whole and see the output. 
( I mean, I could have broken this test case into multiple, and still inject an error in the code that would finally not be caught via the test cases. )  

Anyways, I am open to suggestions if I can make this more readable and easy to go through.